WalletWasabi/Services/WasabiSynchronizer.cs

```diff
@@ -187,18 +187,18 @@ public void Start(TimeSpan requestInterval, TimeSpan feeQueryRequestInterval, in
 								TorStatus = TorStatus.Running;
 								DoNotGenSocksServFail();
 							}
-							catch (TorConnectionException ex)
+							catch (HttpRequestException ex) when (ex.InnerException is TorConnectionException innerEx)
```

Comment:
This `catch` block and the next one are almost identical. It could make sense to unify them as follow:
```suggestion
							catch (HttpRequestException ex) when (ex.InnerException is TorException innerEx)
							{
								TorStatus = innerEx is TorConnectionException ? TorStatus.NotRunning : TorStatus.Running;
								BackendStatus = BackendStatus.NotConnected;
								HandleIfGenSocksServFail(ex);
								HandleIfGenSocksServFail(innerEx);
								throw;
							}
```